### ISSUE TITLE
Commit 9285e9f has broken the build on FreeBSD (i386)

### ISSUE THREAD

**Issue by moisseev** [vstakhov/rspamd#505]
Commit 9285e9f has broken the build on FreeBSD
~~~
# uname -mrs
FreeBSD 10.2-RELEASE i386
~~~

~~~
root@vm-FreeBSD-10_2:/home/moiseev/git # git clone --recursive https://github.com/vstakhov/rspamd.git
Клонирование в <<rspamd>>...
remote: Counting objects: 39623, done.
remote: Compressing objects: 100% (212/212), done.
remote: Total 39623 (delta 132), reused 0 (delta 0), pack-reused 39411
Получение объектов: 100% (39623/39623), 25.04 MiB | 703.00 KiB/s, готово.
Определение изменений: 100% (26332/26332), готово.
Проверка соединения... готово.
root@vm-FreeBSD-10_2:/home/moiseev/git # mkdir rspamd.build
root@vm-FreeBSD-10_2:/home/moiseev/git # cd rspamd.build
root@vm-FreeBSD-10_2:/home/moiseev/git/rspamd.build # cmake ../rspamd
-- The C compiler identification is Clang 3.4.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- The ASM compiler identification is Clang
-- Found assembler: /usr/bin/cc
-- Hyperscan support is possible only for x86_64 architecture
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.28")
-- Found Perl: /usr/local/bin/perl (found version "5.20.3")
-- Performing Test AMD_64
-- Performing Test AMD_64 - Failed
-- Configuring for BSD system
-- Found package LIBUTIL in '/usr/lib' (/usr/lib/libutil.so) and '/usr/include' (libutil.h).
-- Looking for pidfile_open
-- Looking for pidfile_open - found
-- Looking for pidfile_fileno
-- Looking for pidfile_fileno - found
-- Check for luajit 5.1
-- Found luajit 5.1
-- Checking for one of the modules 'glib-2.0>=2.28'
-- Found package GLIB2 in pkg-config modules glib-2.0>=2.28
-- Checking for one of the modules 'pcre;libpcre;pcre3;libpcre3'
-- Found package PCRE in pkg-config modules pcre;libpcre;pcre3;libpcre3
-- Checking for one of the modules 'gmime-2.6;gmime-2.4;gmime-2.0'
-- Found package GMIME in pkg-config modules gmime-2.6;gmime-2.4;gmime-2.0
-- Checking for one of the modules 'event;libevent'
-- Found package LIBEVENT in pkg-config modules event;libevent
-- Checking for one of the modules 'sqlite3;sqlite'
-- Found package SQLITE3 in pkg-config modules sqlite3;sqlite
-- Checking for one of the modules 'iconv'
-- Found package ICONV in '/usr/local/lib' (/usr/local/lib/libiconv.so) and '/usr/include' (iconv.h).
-- Checking for one of the modules 'openssl'
-- Found package OPENSSL in '/usr/lib' (/usr/lib/libcrypto.so) and '/usr/include' (err.h).
-- Checking for one of the modules 'magic'
-- Found package MAGIC in '/usr/lib' (/usr/lib/libmagic.so) and '/usr/include' (magic.h).
-- Performing Test SUPPORT_WALL
-- Performing Test SUPPORT_WALL - Failed
-- Performing Test SUPPORT_W
-- Performing Test SUPPORT_W - Failed
-- Performing Test SUPPORT_WPOINTER
-- Performing Test SUPPORT_WPOINTER - Failed
-- Performing Test SUPPORT_WPARAM
-- Performing Test SUPPORT_WPARAM - Failed
-- Performing Test SUPPORT_WFUNCTION
-- Performing Test SUPPORT_WFUNCTION - Failed
-- Performing Test SUPPORT_WUNUSED_VAR
-- Performing Test SUPPORT_WUNUSED_VAR - Failed
-- Performing Test SUPPORT_WPOINTER_SIGN
-- Performing Test SUPPORT_WPOINTER_SIGN - Failed
-- Performing Test SUPPORT_WSTRICT_PROTOTYPES
-- Performing Test SUPPORT_WSTRICT_PROTOTYPES - Failed
-- Performing Test SUPPORT_PEDANTIC_FLAG
-- Performing Test SUPPORT_PEDANTIC_FLAG - Failed
-- Performing Test SUPPORT_STD_FLAG
-- Performing Test SUPPORT_STD_FLAG - Failed
-- Performing Test SUPPORT_FPIC
-- Performing Test SUPPORT_FPIC - Failed
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for include file sys/time.h
-- Looking for include file sys/time.h - found
-- Looking for include file sys/uio.h
-- Looking for include file sys/uio.h - found
-- Looking for include file fcntl.h
-- Looking for include file fcntl.h - found
-- Looking for include file math.h
-- Looking for include file math.h - found
-- Looking for include file stdio.h
-- Looking for include file stdio.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file string.h
-- Looking for include file string.h - found
-- Looking for include file strings.h
-- Looking for include file strings.h - found
-- Looking for include file time.h
-- Looking for include file time.h - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for include file sysexits.h
-- Looking for include file sysexits.h - found
-- Looking for include file stdint.h
-- Looking for include file stdint.h - found
-- Looking for include file inttypes.h
-- Looking for include file inttypes.h - found
-- Looking for include file stdbool.h
-- Looking for include file stdbool.h - found
-- Looking for include file endian.h
-- Looking for include file endian.h - not found
-- Looking for include file sys/endian.h
-- Looking for include file sys/endian.h - found
-- Looking for include file machine/endian.h
-- Looking for include file machine/endian.h - found
-- Looking for include file sys/queue.h
-- Looking for include file sys/queue.h - found
-- Looking for include file sys/socket.h
-- Looking for include file sys/socket.h - found
-- Looking for include file sys/mman.h
-- Looking for include file sys/mman.h - found
-- Looking for include file sys/un.h
-- Looking for include file sys/un.h - found
-- Looking for include file sys/stat.h
-- Looking for include file sys/stat.h - found
-- Looking for include file sys/wait.h
-- Looking for include file sys/wait.h - found
-- Looking for include file sys/param.h
-- Looking for include file sys/param.h - found
-- Looking for include file sys/cdefs.h
-- Looking for include file sys/cdefs.h - found
-- Looking for include file sys/file.h
-- Looking for include file sys/file.h - found
-- Looking for include file sys/utsname.h
-- Looking for include file sys/utsname.h - found
-- Looking for include file sys/resource.h
-- Looking for include file sys/resource.h - found
-- Looking for include file netinet/in.h
-- Looking for include file netinet/in.h - found
-- Looking for include file arpa/inet.h
-- Looking for include file arpa/inet.h - found
-- Looking for include file netdb.h
-- Looking for include file netdb.h - found
-- Looking for include file syslog.h
-- Looking for include file syslog.h - found
-- Looking for include file siginfo.h
-- Looking for include file siginfo.h - not found
-- Looking for include file locale.h
-- Looking for include file locale.h - found
-- Looking for include file libgen.h
-- Looking for include file libgen.h - found
-- Looking for include file search.h
-- Looking for include file search.h - found
-- Looking for include file pwd.h
-- Looking for include file pwd.h - found
-- Looking for include file grp.h
-- Looking for include file grp.h - found
-- Looking for include file glob.h
-- Looking for include file glob.h - found
-- Looking for include file poll.h
-- Looking for include file poll.h - found
-- Looking for include file readpassphrase.h
-- Looking for include file readpassphrase.h - found
-- Looking for include file termios.h
-- Looking for include file termios.h - found
-- Looking for include file paths.h
-- Looking for include file paths.h - found
-- Looking for include file ctype.h
-- Looking for include file ctype.h - found
-- Looking for include file sys/sendfile.h
-- Looking for include file sys/sendfile.h - not found
-- Looking for include file linux/falloc.h
-- Looking for include file linux/falloc.h - not found
-- Looking for include file sys/eventfd.h
-- Looking for include file sys/eventfd.h - not found
-- Looking for include file aio.h
-- Looking for include file aio.h - found
-- Looking for include file libaio.h
-- Looking for include file libaio.h - not found
-- Looking for include file cpuid.h
-- Looking for include file cpuid.h - found
-- Looking for include file dirent.h
-- Looking for include file dirent.h - found
-- Looking for include file stropts.h
-- Looking for include file stropts.h - not found
-- Looking for include file sys/ioctl.h
-- Looking for include file sys/ioctl.h - found
-- Looking for setproctitle
-- Looking for setproctitle - not found
-- Looking for getpagesize
-- Looking for getpagesize - not found
-- Looking for nanosleep
-- Looking for nanosleep - not found
-- Looking for vfork
-- Looking for vfork - not found
-- Looking for wait4
-- Looking for wait4 - not found
-- Looking for waitpid
-- Looking for waitpid - not found
-- Looking for flock
-- Looking for flock - not found
-- Looking for tanhl
-- Looking for tanhl - not found
-- Looking for tanh
-- Looking for tanh - not found
-- Looking for expl
-- Looking for expl - not found
-- Looking for exp2l
-- Looking for exp2l - not found
-- Looking for sendfile
-- Looking for sendfile - not found
-- Looking for mkstemp
-- Looking for mkstemp - not found
-- Looking for setitimer
-- Looking for setitimer - not found
-- Looking for inet_pton
-- Looking for inet_pton - not found
-- Looking for clock_gettime
-- Looking for clock_gettime - not found
-- Looking for memset_s
-- Looking for memset_s - not found
-- Looking for explicit_bzero
-- Looking for explicit_bzero - not found
-- Performing Test HAVE_WEAK_SYMBOLS
-- Performing Test HAVE_WEAK_SYMBOLS - Failed
-- Performing Test ICONV_SECOND_ARGUMENT_IS_CONST
-- Performing Test ICONV_SECOND_ARGUMENT_IS_CONST - Failed
-- Looking for PATH_MAX
-- Looking for PATH_MAX - not found
-- Looking for MAXPATHLEN
-- Looking for MAXPATHLEN - not found
-- Looking for MAP_SHARED
-- Looking for MAP_SHARED - not found
-- Looking for MAP_ANON
-- Looking for MAP_ANON - not found
-- Looking for MAP_NOCORE
-- Looking for MAP_NOCORE - not found
-- Looking for O_DIRECT
-- Looking for O_DIRECT - not found
-- Looking for IPV6_V6ONLY
-- Looking for IPV6_V6ONLY - not found
-- Looking for posix_fadvise
-- Looking for posix_fadvise - not found
-- Looking for posix_fallocate
-- Looking for posix_fallocate - not found
-- Looking for fallocate
-- Looking for fallocate - not found
-- Looking for fdatasync
-- Looking for fdatasync - not found
-- Looking for _SC_NPROCESSORS_ONLN
-- Looking for _SC_NPROCESSORS_ONLN - not found
-- Looking for setbit
-- Looking for setbit - not found
-- Looking for getaddrinfo
-- Looking for getaddrinfo - not found
-- Looking for sched_yield
-- Looking for sched_yield - not found
-- Looking for __get_cpuid
-- Looking for __get_cpuid - not found
-- Looking for nftw
-- Looking for nftw - not found
-- Looking for PCRE_CONFIG_JIT
-- Looking for PCRE_CONFIG_JIT - not found
-- Looking for SOCK_SEQPACKET
-- Looking for SOCK_SEQPACKET - not found
-- Looking for I_SETSIG
-- Looking for I_SETSIG - not found
-- Looking for O_ASYNC
-- Looking for O_ASYNC - not found
-- Performing Test HAVE_PCRE_JIT_FAST
-- Performing Test HAVE_PCRE_JIT_FAST - Failed
-- pcre_jit_exec is -NOT- supported
-- pthread_mutexattr_setpshared is -NOT- supported
CMake Error at CMakeLists.txt:929 (MESSAGE):
  Your system does not support getaddrinfo call, please consider upgrading it
  to run rspamd


-- Configuring incomplete, errors occurred!
See also "/home/moiseev/git/rspamd.build/CMakeFiles/CMakeOutput.log".
See also "/home/moiseev/git/rspamd.build/CMakeFiles/CMakeError.log".
root@vm-FreeBSD-10_2:/home/moiseev/git/rspamd.build #
~~~

**Comment by vstakhov** [comment]
That should be fixed now, thank you.

**Comment by moisseev** [comment]
Fixed with a8f5d99585a2d8eb23838787b3ab21c331c077d3 Thanks.
